### ISSUE TITLE
Replace Eigen Library with Custom Matrix Multiplication Implementation in C++ Code

### ISSUE THREAD

**Issue by MrNeRF** [MrNeRF/gaussian-splatting-cuda#11]
In our project, we currently utilize the Eigen library for matrix operations. We want to be light on dependencies and we are considering replacing Eigen with our own implementation of matrix multiplication.

So, make it happen and make some tests :)

**Comment by DapengFeng** [comment]
Does it mean to replace Eigen with libtorch?

**Comment by MrNeRF** [comment]
No. I would suggest with own matmul and I believe quaternion ops. Its not much. The goal is to reduce deps from other libs to make it as self contained as possible.

**Comment by MrNeRF** [comment]
Same for glm. There might be overlapping operations with eigen that are used.

**Comment by MrNeRF** [comment]
Are you working on this?

**Comment by DapengFeng** [comment]
> Are you working on this?

Yeah, I will try it.

**Comment by MrNeRF** [comment]
I would still like to remove Eigen.

**Comment by MrNeRF** [comment]
Done
